packages/nimble-components/src/rich-text/editor/tests/rich-text-editor-mention.spec.ts

```diff
@@ -1396,6 +1490,7 @@ describe('RichTextEditorMentionListbox', () => {
             expect(pageObject.getMentionListboxItemsName()).toEqual([]);
             mappingElements[0]!.key = 'user:1';
             await waitForUpdatesAsync();
+            await waitForUpdatesAsync();
```

Comment:
This test case passes only if we add the additional `waitForUpdateAsync()`. We tried to debug the issue and here are our observations:

1. Replicating the same flow in the storybook by updating the DOM opens the mention popup whenever the invalid `key` is changed to a valid key(works as expected with the current implementation):
![invalid-to-valid-state-popup-test-issue](https://github.com/ni/nimble/assets/123377523/357d2156-15fa-41da-88d5-7f68c9f2e07e)

2. When tracking the flow, the `key` of mapping elements changes, we verified that after the first `waitForUpdateAsync()`, the `activeMappingConfigs` updated as expected with the valid mention config data which is responsible for populating the corresponding `nimble-list-option` within mention listbox. However, when we inspect the DOM in the testing environment, there is no `nimble-list-option` within mention listbox. The `activeMappingConfigs` is directly binding to the editor's template but is somehow not reflected in the DOM. Code reference: https://github.com/ni/nimble/blob/8ff070c7058f1602444a010db1dcc324016ff37f/packages/nimble-components/src/rich-text/editor/template.ts#L120C13-L122C36
![image](https://github.com/ni/nimble/assets/123377523/73ee1d16-bded-445c-8d7c-cfc59410c990)

After the second `waitForUpdatesAsync()`, the DOM updates with the expected `nimble-list-option` and opens the mention popup.

Reply:
I think that's okay. Please just add a comment explaining that after the first wait, `activeMappingConfigs` is updated, and after the second wait, the `nimble-list-options` get populated.

Reply:
Added a comment explaining the rationale of two awaits as suggested.